### PR TITLE
more form filter cleanup

### PR DESCRIPTION
**Pull request by czue**
:blowfish: (sorry the last one's pretty big)

@snopoke @benrudolph

### CODE REVIEW

File: corehq/apps/reports/filters/forms.py [hunk 1]
```diff
@@ -4,8 +4,10 @@
 from django.utils.safestring import mark_safe
 import re
 from corehq.apps.app_manager.models import Application
-from corehq.apps.reports.analytics.couchaccessors import guess_form_name_from_submissions_using_xmlns
+from corehq.apps.reports.analytics.couchaccessors import guess_form_name_from_submissions_using_xmlns, \
+    get_all_form_definitions_grouped_by_app_and_xmlns, get_all_app_structures
 from corehq.apps.reports.filters.base import BaseDrilldownOptionFilter, BaseSingleOptionFilter, BaseTagsFilter
+from corehq.util.debug import print_return_value
```
Comment (dimagimon) [inline corehq/apps/reports/filters/forms.py]: F401 'print_return_value' imported but unused

**Comment by czue** [comment]
don't merge since http://manage.dimagi.com/default.asp?189029 likely means that what this builds on needs to get reverted

File: corehq/apps/reports/filters/forms.py [hunk 2]
```diff
@@ -1,11 +1,12 @@
-import copy
 from couchdbkit.exceptions import ResourceNotFound
 from django.conf import settings
 from django.utils.safestring import mark_safe
-import re
 from corehq.apps.app_manager.models import Application
-from corehq.apps.reports.analytics.couchaccessors import guess_form_name_from_submissions_using_xmlns
+from corehq.apps.reports.analytics.couchaccessors import guess_form_name_from_submissions_using_xmlns, \
```
Comment (dimagimon) [inline corehq/apps/reports/filters/forms.py]: F401 'FormDetails' imported but unused

File: corehq/apps/reports/filters/forms.py [hunk 3]
```diff
@@ -409,66 +407,96 @@ def _get_filtered_data(self, filter_results):
...
         parsed_params = FormsByApplicationFilterParams(filter_results)
         if parsed_params.xmlns:
-            status = parsed_params.status or PARAM_VALUE_STATUS_ACTIVE
-            key = ["status xmlns app", self.domain, status, parsed_params.xmlns, parsed_params.app_id]
-            return self._raw_data(key)
+            return get_form_details_for_app_and_xmlns(
+                self.domain,
+                parsed_params.app_id,
+                parsed_params.xmlns,
+                deleted=parsed_params.status==PARAM_VALUE_STATUS_DELETED,
```
Comment (dimagimon) [inline corehq/apps/reports/filters/forms.py]: E225 missing whitespace around operator

**Comment by czue** [comment]
i think this is good to go, though we might want to do another morning merge and india deploy just in case any unexpected issues

**Comment by czue** [comment]
also not sure how best to review. is broken out but i took a number of detours. maybe I can just walk you through it?

**Comment by snopoke** [comment]
Phew, took a quick look but would be great to get a walk through.

**Comment by benrudolph** [comment]
+1

**Pull request merged**